tests/e2e/components/kubectl-canvas.ts

```diff
@@ -0,0 +1,49 @@
+import { expect, Locator, Page } from '@playwright/test'
+
+import { step } from '../rancher/rancher-test'
+import { BaseShell } from './kubectl-base'
+
+export class Shell extends BaseShell {
+    readonly canvas: Locator
+
+    constructor(page: Page) {
+      super(page)
+      this.canvas = this.screen.locator('canvas.xterm-link-layer')
+    }
+
+    @step
+    async run(cmd: string, options?: { status?: number, inPlace?: boolean, timeout?: number }): Promise<number> {
+      const status = options?.status ?? 0
+      const timeout = options?.timeout || 60_000
+
+      if (!options?.inPlace) await this.open()
+
+      // Fill is faster but removes newlines, multiline commands require input.pressSequentially
+      await this.cursor.fill(`${cmd}; ES=$?; clear; echo EXITSTATUS-$ES`)
+      await this.cursor.press('Enter')
+
+      const box = await this.canvas.boundingBox() || { x: 72, y: 481 }
+      const clip = { x: box.x, y: box.y, width: 100, height: 14 }
+
+      // Command finished when we see EXITSTATUS-
+      await expect(async() => {
+        await expect(this.page).toHaveScreenshot('shell/EXITSTATUS-NaN.png', { clip: { ...clip, width: 77 } })
```

Comment:
Will the canvas always have the `EXITSTATUS-...` in the same location within the window?

Reply:
Yes, the trick is to [clear terminal after running command](https://github.com/rancher/kubewarden-ui/pull/608/files#diff-d310c122126635379c842d459229e76d5047e6f4b15ecec186a6f6f2cbcc69d8R22), so it would contain only `EXITSTATUS-...` 

For example `shell.run('ls')` would produce:
```bash
ls; ES=$?; clear; echo EXITSTATUS-$ES
```

Reply:
Ah okay, thanks for the clarification.